kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderRetransmitDocument.java

```diff
@@ -0,0 +1,204 @@
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForApprovedPODocuments(this);
+                setPurchaseOrderLastTransmitTimestamp(SpringContext.getBean(DateTimeService.class).getCurrentTimestamp());
+                updateAndSaveAppDocStatus(PurchaseOrderStatuses.APPDOC_OPEN);
+            }
+            // DOCUMENT DISAPPROVED
+            else if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isDisapproved()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForDisapprovedChangePODocuments(this);
+
+                // for app doc status
+                try {
+                    String nodeName = SpringContext.getBean(WorkflowDocumentService.class).getCurrentRouteLevelName(this.getFinancialSystemDocumentHeader().getWorkflowDocument());
+                    String reqStatus = PurapConstants.PurchaseOrderStatuses.getPurchaseOrderAppDocDisapproveStatuses().get(nodeName);
+                    updateAndSaveAppDocStatus(PurapConstants.PurchaseOrderStatuses.getPurchaseOrderAppDocDisapproveStatuses().get(reqStatus));
+                } catch (WorkflowException e) {
+                    logAndThrowRuntimeException("Error saving routing data while saving App Doc Status " + getDocumentNumber(), e);
+                }
+            }
+            // DOCUMENT CANCELED
+            else if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isCanceled()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForCancelledChangePODocuments(this);
+                // for app doc status
+                updateAndSaveAppDocStatus(PurapConstants.PurchaseOrderStatuses.APPDOC_CANCELLED);
+            }
+        }
+        catch (WorkflowException e) {
+            logAndThrowRuntimeException("Error saving routing data while saving document with id " + getDocumentNumber(), e);
+        }
+    }
+
+    public boolean isShouldDisplayRetransmitTab() {
+        return shouldDisplayRetransmitTab;
+    }
+
+    public void setShouldDisplayRetransmitTab(boolean shouldDisplayRetransmitTab) {
+        this.shouldDisplayRetransmitTab = shouldDisplayRetransmitTab;
+    }
+
+    @Override
+    protected boolean shouldAdhocFyi() {
+        return false;
+    }
+    
+    /**
+     * Gets the routeCodeObj attribute. 
+     * @return Returns the routeCodeObj.
+     */
+    public Building getBuildingObj() {
+        return buildingObj;
+    }
+
+    /**
+     * Sets the routeCodeObj attribute value.
+     * @param routeCodeObj The routeCodeObj to set.
+     */
+    public void setBuildingObj(Building buildingObj) {
+        this.buildingObj = buildingObj;
+    }
+    
+    /**
+     * @see org.kuali.kfs.module.purap.document.PurchasingDocument#getRouteCode()
+     */
+    public String getRouteCode() {
+    	
+    	try{
```

Comment:
Should only retrieve route code from Building extension if route code is blank:
if (StringUtils.isBlank(routeCode)) {

try {
BuildingExtension be = (BuildingExtension) buildingObj.getExtension();